backend/server.js

```diff
@@ -22,4 +21,57 @@ function authenticateToken(req, res, next) {
   });
 }
 
-app.listen(5000);
+// Create a new post
+app.post("/posts", authenticateToken, async (req, res) => {
+  const { title, content, imageUrls } = req.body;
+
+  try {
+    const newPost = await prisma.post.create({
+      data: {
+        title,
+        content,
+        user: { connect: { username: req.user.name } },
+        images: {
+          create: imageUrls.map((url) => ({ url })),
+        },
+      },
+      include: { images: true },
+    });
+
+    res.status(201).json(newPost);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: "Error creating post", error: error.message });
+  }
+});
+
+// Get all posts from all users
+app.get("/allposts", authenticateToken, async (req, res) => {
+  try {
+    const posts = await prisma.post.findMany({
+      include: {
+        images: true,
+        user: {
+          select: {
+            username: true,
+            // Add other user fields want to include, but NOT the password
+          },
+        },
+      },
+      orderBy: {
+        createdAt: "desc", // This will sort posts from newest to oldest
+      },
+    });
+
+    res.json(posts);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: "Error fetching all posts", error: error.message });
+  }
+});
+
+app.listen(PORT, () => {
+  console.log(`Server is running on http://localhost:${PORT}`);
```

Comment:
No need to mention which port the website is running on; Security POV

Reply:
Just went back and removed it from both files and will keep this in mind.